### ISSUE TITLE
Add pagination in landingzones file list modal

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-server#1881]
@Nicolai-vKuegelgen recently did a bit of stress testing on the landing zones in production. As a result, I've encountered many interesting things which probably should've been thought of while developing. See also #1872.

Suffice to say, it turns out that attempting to retrieve and display a flat list of **170 000 files** does not result in a decent outcome with the current implementation :)

We should add pagination to the modal to prevent attempting to load instane amount of files in batch.

Filtering should also be added, but that's a different ticket.

Related to #1730, which should be done first.

Update: There are also some other issues I failed to reconsider.

- `IrodsAPI.get_objects()` can be used to fetch .md5 checksum files, but those count in the limit for the file count. Hence we'll probably need to fetch the .md5 availability information in a separate Ajax request.
- `IrodsAPI.get_object_stats()` doesn't take collections into account, so retrieving stats for total amount needed for pagination controls will not work with the current implementation. See #2160.
- `IrodsAPI.get_objects()` does not actually work with `include_colls=True` and `limit`/`offset` params. See #2159.

## Spec

- Do #2159 
- Do #2160
- Add Django setting for pagination size
    * Remember to add this to siteinfo
- Add pagination support in `ZoneIrodsListRetrieveAjaxView`
- Add pagination controls in zone file list modal UI
- Add .md5 availability Ajax view
- Add .md6 Ajax view calling in modal JQuery

## Tasks

- [x] Do #2159 
- [x] Do #2160
- [x] Add Django setting for pagination
- [x] Add setting to siteinfo
- [x] Add pagination support for Ajax view
- [x] Update Ajax view tests for pagination
- [x] Add pagination controls in UI
- [x] Improve modal CSS
- [ ] ~~Update UI tests for pagination~~ See #2167, #2168
- [x] Add separate MD5 status retrieval Ajax view
- [x] Add MD5 status Ajax view tests
- [x] Add MD5 status retrieval controls
- [ ] ~~Add MD5 status UI tests~~ See above
- [x] Update documentation

**Comment by mikkonie** [comment]
Postponed this to v1.1, to be done together with #1996 and #1997.

**Comment by mikkonie** [comment]
I recognized some IrodsAPI issues which need to be solved before this feature can reasonably be implemented. I'll see if doing all this is viable for v1.1, otherwise postponing to v1.2.

**Comment by mikkonie** [comment]
Turns out #2159 and #2160 were easy to do, albeit the implementation for the former is not necessary an optimal one (see #1883 for more). Regardless, this can now be done for v1.1 as it was intended.

**Comment by mikkonie** [comment]
Sadly, UI tests for the modal need to be postponed for now. I was wondering why they're not there, but turns out it's because we don't have nice base classes or even working helpers for Taskflow-enabled UI testing. See #2167, #2168.

**Comment by mikkonie** [comment]
Done. Missing UI tests need to be added later once the test framework is appropriately tweaked, see #2169.
